### ISSUE TITLE
[Question] Auto-merging Pin PRs

### ISSUE THREAD

**Issue by keplersj** [renovatebot/config-help#185]
### Which Renovate are you using? CLI, App, or Pro
Renovate App

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps
GitHub

### Have you checked the logs? Don't forget to include them if relevant

Yes, it looks like Renovate recognizes that the PR is configured for auto merge, but has determined the PR isn't ready for some reason.

```
DEBUG: Pull Request #12 does not need updating(dependencies=["semantic-release","semantic-release-chrome","semantic-release-firefox"], branch="renovate/pin-dependencies")
DEBUG: Checking #12 for automerge(dependencies=["semantic-release","semantic-release-chrome","semantic-release-firefox"], branch="renovate/pin-dependencies")
{
  "automerge": true,
  "automergeType": "pr",
  "automergeComment": "automergeComment"
}
DEBUG: PR is configured for automerge(dependencies=["semantic-release","semantic-release-chrome","semantic-release-firefox"], branch="renovate/pin-dependencies")
INFO: PR is not ready for merge(dependencies=["semantic-release","semantic-release-chrome","semantic-release-firefox"], branch="renovate/pin-dependencies")
```

[Link to Log](https://renovatebot.com/dashboard#github/keplersj/cises/52019033)

### What is your question?

Why aren't my pin pull requests auto-merging? I've observed patch PRs auto-merge as configured in the repo, and I'm not sure why pin PRs aren't doing the same.

[Repo](https://github.com/keplersj/cises)
[Pull Request](https://github.com/keplersj/cises/pull/12)

**Comment by rarkins** [comment]
When automerging, Renovate makes sure the PR is up to date with master first, and you have made commits to master that keep causing the PR to be rebased. You can see the description in the PR:

![image](https://user-images.githubusercontent.com/6311784/54311782-9e6ffa00-45d5-11e9-9a66-3f7d45b02b57.png)

Also, automerging happens during a normal run and isn't triggered by the branch going green (it's a long story, but GitHub's webhooks on this aren't very good so we can't rely on them).

In short: Renovate will automerge any time it runs if (a) tests pass, and (b) the PR remains up to date with master. Otherwise it rebases and starts over.
